TERRAFORM.md

```diff
@@ -143,6 +144,185 @@ Many tests require access to a Kubernetes cluster, such as a local [KinD cluster
+resource "imagetest_harness_k3s" "foo" {
+  # The inventory this harness should be registered against. This is primarily
+  # used for internal provider plumbing, and as such, the data format of the
+  # input is explicitly the full data structure of `imagetest_inventory` (and not
+  # the ID) to make it harder to mess up.
+  inventory = data.imagetest_inventory.this
+
+  # A required harness name. This must be unique within the scope of the
+  # provided inventory. For example, defining another named "foo" harness with
+  # the same `data.imagetest_inventory.this` is invalid, but another "foo"
+  # harness with `data.imagetest_inventory.that` is valid and often normal,
+  # such as when working with for_each'd modules.
+  name = "foo"
+}
+
+# All tests run within a `imagetest_feature` resource, and are a set of
+# instructions that run to completion. The way they are run and how they are run
+# vary depending on the provided harness.
+resource "imagetest_feature" "foo" {
+  # A feature must have a harness, and much like the required inventory field
+  # in the harness, this field is structured in a way to take the full data
+  # structure of the harness (and not the ID) to make it harder to mess up. The
+  # harness passed to the feature dictates _how_ the steps in the feature are
+  # executed. These are documented in each of the available imagetest_harness_*
+  # resources
+  harness = imagetest_harness_k3s.foo
+
+  # The name of the feature. This must be unique within a given harness.
+  name = "Footure"
+  # The optional description of the harness
+  description = "My great footure"
+
+  # Steps are a series of commands executed in order to completion. Depending #
+  # on the provided harness, these location and method of how these steps are #
+  # executed will vary, but in general, they are typically abstracted to operate
+  # # as shell scripts. For example, in this example, the `imagetest_harness_k3s`
+  # is used, which means these steps will execute within a sandboxed `wolfi-base`
+  # image that has `kubectl` pre-installed and pre-configured to connect to the
+  # network attached k3s cluster. The intent with the harness feature setups is
+  # always to absolve as boilerplate setup as possible from the test authors.
+  steps = [
+    {
+      name = "Do a thing"
+      cmd = <<EOF
+        kubectl get po -A
+      EOF
+    },
+    # Where applicable, additional step fields are available to hoist commonly
+    # performed actions out of shell scripts and into the provider. The example
+    # below shows an optional retry = {} block, which abstracts the common
+    # approach of a bash retry loop with a simple provider resource definition
+    {
+      name = "Do a thing that may flake"
+      cmd = <<EOF
+        ...
+      EOF
+      # Retry with a linear backoff
+      retry = { attempts = 5, delay = "10s" },
+      # Retry with an exponential backoff
+      # retry = { attempts = 3, delay = "5s", factor = "2" },
```

Comment:
This can be a little confusing to read when a separate example is commented out:
```suggestion
      # Retry with a linear backoff
      retry = { attempts = 5, delay = "10s" },
      # Retry with an exponential backoff
      retry = { attempts = 3, delay = "5s", factor = "2" },
```

Perhaps break it down into two examples?